nixos/tests/clevis.nix

```diff
@@ -0,0 +1,440 @@
+      { config, lib, pkgs, ... }:
+
+      {
+        imports = [
+          ./hardware-configuration.nix
+          <nixpkgs/nixos/modules/testing/test-instrumentation.nix>
+        ];
+
+        networking.hostName = "installed";
+
+        boot.loader.grub = {
+          extraConfig = "serial; terminal_output serial";
+          device = "/dev/vda";
+          fsIdentifier = "uuid";
+          configurationLimit = 100;
+        };
+
+
+        boot.kernelParams = [ "console=tty0" ${optionalString (!systemdStage1)  ''"ip=192.168.1.1:::255.255.255.0::eth1:none"''} ];
+
+        boot.initrd = {
+          availableKernelModules = [ "tpm_tis" ];
+          clevis = { enable = true; useTang = true; };
+          ${optionalString (!systemdStage1) "network.enable = true;"}
+          ${optionalString systemdStage1 ''
+            systemd = {
+              enable = true;
+              network.enable = true;
+              network.networks."10-static" = {
+                matchConfig.Name = [ "en*" "eth*" ];
+                DHCP = "no";
+                address = [ "192.168.1.1/24" ];
+                linkConfig.RequiredForOnline = true;
+                networkConfig.IPv6PrivacyExtensions = "kernel";
+              };
+            };
+          ''}
+        };
+
+        # To ensure that we can rebuild the grub configuration on the nixos-rebuild
+        system.extraDependencies = with pkgs; [ stdenvNoCC ];
+        documentation.enable = false;
+        hardware.enableAllFirmware = lib.mkForce false;
+        networking.interfaces."eth1".useDHCP = false;
+        ${replaceStrings [ "\n" ] [ "\n  " ] extraConfig}
+      }
+    '';
+
+  testScriptFun = { createRootPartition, extraConfig, systemdStage1, fallback, preBootCommands }:
+    let
+      createQemuMachine = ''
+        import os
+
+        # Set the QEMU flags to create the machine where we install NixOS
+        def assemble_qemu_flags():
+            flags = "-cpu max"
+            ${if (system == "x86_64-linux" || system == "i686-linux") then ''
+              flags += " -m 1024"
+            '' else ''
+              flags += " -m 768 -enable-kvm -machine virt,gic-version=host"
+            ''}
+            flags += " -chardev socket,id=chrtpm,path=/tmp/swtpm-sock -tpmdev emulator,id=tpm0,chardev=chrtpm -device tpm-tis,tpmdev=tpm0"
+            flags += " -device virtio-net-pci,netdev=vlan1,mac=52:54:00:12:11:02 -netdev vde,id=vlan1,sock=\"$QEMU_VDE_SOCKET_1\""
+            return flags
+
+
+        qemu_flags = {"qemuFlags": assemble_qemu_flags()}
+
+        image_dir = installer.state_dir
+        disk_image = os.path.join(image_dir, "installer.qcow2")
+
+        hd_flags = {
+            "hdaInterface": "virtio",
+            "hda": disk_image,
+        }
+
+        default_flags = {**hd_flags, **qemu_flags}
+
+        def create_machine_named(name):
+            return create_machine({**default_flags, "name": name})
+      '';
+
+      tpmEmulator = ''
```

Comment:
Does #229035 help with this?

Reply:
Not really, I've tried. The problem here is that the tpm has to persist to the second vm that is created via `create_machine_named`. I am not sure but I think that with #229035 the `swtpm` socket is destroyed after the installer shutdowns.